pkg/systemd/overview-cards/cryptoPolicies.jsx

```diff
@@ -0,0 +1,199 @@
+
+import { ModalError } from 'cockpit-components-inline-notification.jsx';
+import { PrivilegedButton } from "cockpit-components-privileged.jsx";
+import { TunedDialogBody } from '../../../pkg/tuned/change-profile.jsx';
+
+import './cryptoPolicies.scss';
+
+const _ = cockpit.gettext;
+
+// From /usr/share/crypto-policies/policies/$POLICY.pol
+const cryptopolicies = {
+    DEFAULT: _("A reasonable default for today's standards. It should provide 112-bit security with the exception of SHA1 signatures in DNSSec SHA1 is also enabled in HMAC where collision attacks do not matter."),
+    FUTURE: _("A level that will provide security on a conservative level that is believed to withstand any near-term future attacks. And also provide some (not complete) preparation for post quantum encryption support in form of 256 bit symmetric encryption requirement. It provides at least an 128-bit security. This level may prevent communication with many used systems that provide weaker security levels (e.g., systems that use SHA-1 as signature algorithm)."),
+    LEGACY: _("Provides settings for ensuring maximum compatibility with legacy systems. This policy is less secure and intended to be a easy way to switch system to be compatible with older systems. It should provide at least 80-bit security and excludes 3DES and RC4."),
+};
+const defaultpolicy = "DEFAULT";
+const stampfile = "/run/cockpit/crypto-policies-reboot-stamp";
+
+export const CryptoPolicyRow = () => {
+    const [showCryptoPolicy, setShowCryptoPolicy] = useState(false);
+    const [currentCryptoPolicy, setCurrentCryptoPolicy] = useState("");
+    const [isOpen, setIsOpen] = useState();
+
+    useEffect(() => {
+        // Avoid cockpit.spawn every re-render from parent
+        if (!showCryptoPolicy) {
+            cockpit.spawn(["update-crypto-policies", "--show"]).then(output => {
+                setCurrentCryptoPolicy(output.trim());
+                setShowCryptoPolicy(true);
+            });
+        }
+    }, [showCryptoPolicy]);
+
+    if (!showCryptoPolicy) {
+        return null;
+    }
+    const permission = cockpit.permission({ admin: true });
+
+    return (
+        <tr>
+            <th scope="row">{_("Crypto policy")}</th>
+            <td>
+                <PrivilegedButton variant="link" buttonId="crypto-policy-button" tooltipId="tip-cpu-security"
+                                excuse={ _("The user $0 is not permitted to change crypto policies") }
+                                permission={ permission } onClick={() => setIsOpen(true)}>
+                    {currentCryptoPolicy}
+                </PrivilegedButton>
+                {isOpen &&
+                <CryptoPolicyDialog close={() => setIsOpen(false)}
+                          currentCryptoPolicy={currentCryptoPolicy}
+                          setCurrentCryptoPolicy={setCurrentCryptoPolicy}
+                />
+                }
+            </td>
+        </tr>
```

Comment:
Why are we using tables here? Why aren't we using the PF menu item, like in languages (at least I think languages is now using it)?

I think that a plain, scrollable menu is probably what's wanted? And probably with a border, so it's obvious it's a widget it's scrollable.

https://www.patternfly.org/v4/components/menu#with-description
https://www.patternfly.org/v4/components/menu/html#plain (`isPlain` and `isScrollable` in React, although there's no demo for it in React)

I don't see a helper to add a border, oddly. That seems like an omission in PF. I suppose we could do something in custom CSS like:

```css
border: var(--pf-global--BorderWidth--sm) solid var(--pf-global--BorderColor--100)
```

Reply:
This is the overview card which always used a table and we can't really change all the overview cards in this PR but we can make an issue to convert it in a separate PR.